File: benchmarks/hash_table/hash_table.py
import sys
import time

n = int(sys.argv[1]) if len(sys.argv) > 1 else 100000

start = time.perf_counter()

ht = {}
checksum = 0

# Insert n items
for i in range(n):
    ht[f"key_{i}"] = i * 2

# Lookup all items
for i in range(n):
    key = f"key_{i}"
    if key in ht:
        checksum += ht[key]

# Delete half the items
for i in range(0, n, 2):
    del ht[f"key_{i}"]

# Lookup remaining items
for i in range(n):
    key = f"key_{i}"
    if key in ht:
        checksum += ht[key]

elapsed = (time.perf_counter() - start) * 1000
print(f"TIME_MS:{elapsed:.2f}", file=sys.stderr)
print(checksum)

File: benchmarks/hash_table/hash_table.rb
n = ARGV[0] ? ARGV[0].to_i : 100000

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)

ht = {}
checksum = 0

# Insert n items
n.times do |i|
  ht["key_#{i}"] = i * 2
end

# Lookup all items
n.times do |i|
  key = "key_#{i}"
  checksum += ht[key] if ht.key?(key)
end

# Delete half the items
(0...n).step(2) do |i|
  ht.delete("key_#{i}")
end

# Lookup remaining items
n.times do |i|
  key = "key_#{i}"
  checksum += ht[key] if ht.key?(key)
end

elapsed = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000
STDERR.puts "TIME_MS:#{format('%.2f', elapsed)}"
puts checksum
